python/cudf/cudf/core/indexed_frame.py

```diff
@@ -867,3 +868,62 @@ def resample(
             if isinstance(self, cudf.Series)
             else cudf.core.resample.DataFrameResampler(self, by=by)
         )
+
+    def _reset_index(self, level, drop, col_level=0, col_fill=""):
+        """Shared path for DataFrame.reset_index and Series.reset_index."""
+        if col_level != 0:
+            raise NotImplementedError(
+                "col_level parameter is not supported yet."
+            )
+
+        if col_fill != "":
+            raise NotImplementedError(
+                "col_fill parameter is not supported yet."
+            )
+
+        if level is not None and not isinstance(level, (tuple, list)):
+            level = (level,)
+        _check_duplicate_level_names(level, self._index.names)
+
+        # Split the columns in the index into data and index columns
+        (
+            data_columns,
+            index_columns,
+            column_names,
+            index_names,
+        ) = self._index._split_columns_by_levels(level)
+        if index_columns:
+            index = _index_from_data(
+                dict(zip(range(len(index_columns)), index_columns)),
+                name=self._index.name,
+            )
+            if isinstance(index, MultiIndex):
+                index.names = index_names
+            else:
+                index.name = index_names[0]
+        else:
+            index = RangeIndex(len(self))
+
+        if drop:
+            return self._data, index
+
+        new_column_data = {}
+        for name, col in zip(column_names, data_columns):
+            if name == "index" and "index" in self._data:
+                name = "level_0"
+            new_column_data[name] = col
+        return {**new_column_data, **self._data}, index
+
+
+def _check_duplicate_level_names(specified, level_names):
+    if specified is None:
+        return
+    if len(set(level_names)) == len(level_names):
+        return
+    duplicates = {key for key, val in Counter(level_names).items() if val > 1}
+
+    for x in specified:
+        if x in duplicates:
+            raise ValueError(
+                f"The name {x} occurs multiple times, use a level number"
```

Comment:
Would it be more useful to collect all of the duplicate names and then print them all in the error message?

Reply:
Sure. Pandas doesn't do that but I think we can use discretion here.

Reply:
Up to you, this isn't a must. Just a suggestion since otherwise the user has to go through multiple rounds of fixes if they make the same mistake with multiple duplicate level names.

Reply:
done